src/nspawn/nspawn-network.c

```diff
@@ -99,10 +101,32 @@ static int generate_mac(
         return 0;
 }
 
+static int set_alternative_ifname(sd_netlink *rtnl, const char *ifname, const char *altifname) {
+        _cleanup_strv_free_ char **a = NULL;
+        int r;
+
+        assert(rtnl);
+        assert(ifname);
+
+        if (!altifname)
+                return 0;
+
+        r = strv_extend(&a, altifname);
+        if (r < 0)
+                return log_oom();
+
+        r = rtnl_set_link_alternative_names_by_ifname(&rtnl, ifname, a);
```

Comment:
we can declare an strv on the stack here, via `STRV_MAKE(altifname)`... then we don't have to use `a` at all, and no strv_extend()